source/components/Select/index.jsx

```diff
@@ -0,0 +1,200 @@
+        items[matchIndex].text.toLowerCase() < newValue.toLowerCase()
+      ) {
+        matchIndex += 1;
+      }
+
+      document.getElementById(this.id).scrollTop = 37 * matchIndex;
+
+      this.setState({ inputValue: newValue, matchIndex });
+    }
+  }
+
+  handleKeyDown(e) {
+    const { isOpen } = this.state;
+    if (isOpen) {
+      if (e.key === 'ArrowUp') {
+        let { matchIndex } = this.state;
+        const { items } = this.props;
+        if (matchIndex !== 0) {
+          matchIndex -= 1;
+          if (document.getElementById(this.id).scrollTop >= 37 * (matchIndex + 1)) {
+            document.getElementById(this.id).scrollTop -= 37;
+          }
+          this.setState({ matchIndex, inputValue: items[matchIndex].text });
+        }
+      } else if (e.key === 'ArrowDown') {
+        let { matchIndex } = this.state;
+        const { items } = this.props;
+        if (matchIndex !== items.length - 1) {
+          matchIndex += 1;
+          if (document.getElementById(this.id).scrollTop <= 37 * (matchIndex - 5)) {
+            document.getElementById(this.id).scrollTop += 37;
+          }
+          this.setState({ matchIndex, inputValue: items[matchIndex].text });
+        }
+      } else if (e.key === 'Enter' || e.key === 'Tab') {
+        const { items, onSelect } = this.props;
+        const { matchIndex } = this.state;
+        this.setState({
+          curIndex: matchIndex,
+          inputValue: items[matchIndex].text,
+          isOpen: false,
+        });
+        onSelect(matchIndex);
+      }
+    } else {
+      if (e.key === 'Enter') {
+        e.target.select();
+        this.setState({ isOpen: true });
+      }
+    }
+  }
+
+  handleFocus(e) {
+    e.preventDefault();
+    this.toggleMenu();
+    this.inputRef.current.focus();
+  }
+
+  render() {
+    const { items, label } = this.props;
+    const { curIndex, isOpen, inputValue, matchIndex } = this.state;
+
+    const menuHeight = items.length * 37 < 185 ? items.length * 37 : 185;
+    const openedStyling = {
+      height: menuHeight,
+    };
+    const closedStyling = {
+      height: 0,
+      visibility: 'hidden',
+    };
+
+    return (
+      <div className="select-cont">
+        <div className="select-label">{label}</div>
+        <div className="select-input-cont" onClick={this.handleFocus}>
+          <input
+            className={curIndex === -1 ? 'select-input' : 'select-input select-dark-text'}
+            value={inputValue}
+            onChange={this.handleChange}
+            onKeyDown={this.handleKeyDown}
+            ref={this.inputRef}
+          />
+          <img
+            className={isOpen ? 'select-drop-down-arrow select-rotated' : 'select-drop-down-arrow'}
+            src={dropDownArrow}
+            alt="Drop down arrow"
+          />
+        </div>
+        <div className="select-menu" style={isOpen ? openedStyling : closedStyling} id={this.id}>
+          {items.map((item, index) => (
+            <div
```

Comment:
Text should be wrapped in an HTML element semantically meant for text like `<p>`